qiskit/transpiler/passes/optimization/commutative_inverse_cancellation.py

```diff
@@ -0,0 +1,91 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Cancel pairs of inverse gates exploiting commutation relations."""
+
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class CommutativeInverseCancellation(TransformationPass):
+    """Cancel pairs of inverse gates exploiting commutation relations."""
+
+    def _skip_node(self, node):
+        """Returns True if we should skip this node for the analysis."""
+        if node.op._directive or node.name in {"measure", "reset", "delay"}:
+            return True
+        if node.op.condition:
+            return True
+        # ToDo: Not sure about the next line
```

Comment:
I think this is fine, unless you know for a parameterized gate if the commutative and inverse checking will eval true even with a `Parameter` or `ParameterExpression` object are used for any gate parameters.

Reply:
We probably want to have this extension in the future, but let's keep things simple for now. This comment is also relevant to many other transpiler passes, which usually also ignore conditional gates, parameterized gates, and directives (e.g. collecting 1q or 2q runs).